aws/table_aws_eks_cluster.go

```diff
@@ -0,0 +1,192 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
+func tableAwsEksCluster(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_eks_cluster",
+		Description: "AWS Elastic Kubernetes Service Cluster",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("name"),
+			Hydrate:    getEksCluster,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listEksClusters,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the cluster.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "created_at",
+				Description: "The Time when the cluster was created.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "version",
+				Description: "The Kubernetes server version for the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "endpoint",
+				Description: "The endpoint for your Kubernetes API server.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "role_arn",
+				Description: "The Amazon Resource Name (ARN) of the IAM role that provides permissions for the Kubernetes control plane to make calls to AWS API operations on your behalf.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "resources_vpc_config",
+				Description: "The VPC configuration used by the cluster control plane.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "logging",
+				Description: "The logging configuration for the cluster.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "identity",
+				Description: "The identity provider information for the cluster.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "status",
+				Description: "The current status of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "certificate_authority",
+				Description: "The certificate-authority-data for the cluster.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "platform_version",
+				Description: "The platform version of your Amazon EKS cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
```

Comment:
There are few more properties like `KubernetesNetworkConfig` and `EncryptionConfig` which are not added.